src/Orbits.jl

```diff
@@ -1,194 +1,385 @@
+    if m_1f == -1
+        m_1f = m_1i
+    end
+    M_i = (m_1i+m_2i)*m_sun  
+    M_f = (m_1f+m_2f)*m_sun 
 
-function post_kick_parameters_a_e(a,e,m1,m2,cosν,sinν,vkick,
-    cosθ, sinθ, cosϕ, sinϕ,vim, m1f, m2f,
-    cosΩ, sinΩ, cosω, sinω, cosi, sini)
+    # convert trig functions to vars
+    cosθ = cos(θ)
+    sinθ = sin(θ)
+    cosϕ = cos(ϕ)
+    sinϕ = sin(ϕ)
+    cosν_i = cos(ν_i)
+    sinν_i = sin(ν_i)
+    cosi_i = cos(i_i)
+    sini_i = sin(i_i)
+    cosω_i = cos(ω_i)
+    sinω_i = sin(ω_i)
+    cosΩ_i = cos(Ω_i)
+    sinΩ_i = sin(Ω_i)
+
+    # construct useful intermediary parameters
+    f_ν = (1 - e_i^2)/(1 + e_i*cosν_i)
+    g_ν = sqrt((1 + 2*e_i*cosν_i + e_i^2)/(1 - e_i^2))
 
-    f_ν = (1-e^2)/(1+e*cosν)
-    g_ν = sqrt((1+2*e*cosν+e^2)/(1-e^2))
-    M = m1+m2
-    Mf = m1f+m2f
-    vrel = g_ν*sqrt(cgrav*M/a)
+    v_rel = g_ν*sqrt(cgrav*M_i/a_i)/(km)
 
-    h_ν = -e*sinν/sqrt(1+2*e*cosν+e^2)
-    j_ν = (1+e*cosν)/sqrt(1+2*e*cosν+e^2)
+    h_ν = -e_i*sinν_i/sqrt(1 + 2*e_i*cosν_i + e_i^2)
+    j_ν = (1 + e_i*cosν_i)/sqrt(1 + 2*e_i*cosν_i + e_i^2)
 
-    α = vkick/vrel
-    β = vim/vrel
+    α = vkick/v_rel
+    β = vimp/v_rel
 
-    ξ = f_ν*g_ν^2*M/Mf*(1+α^2+β^2+2*(-h_ν*β*(1+α*cosθ)+α*cosθ-j_ν*β*α*sinθ*cosϕ))
+    #println("=> ", f_ν, " ", g_ν, " ", h_ν, " ", j_ν, " ")
+    #println("==> ", cosθ, " ", cosϕ)
+    ξ = f_ν*g_ν^2*M_i/M_f* (1 + α^2 + β^2 + 2* (α*cosθ - h_ν*β* (1 + α*cosθ) - j_ν*β*α*sinθ*cosϕ))
+    #println(ξ)
 
     if ξ>2
-        return (NaN, NaN, NaN, NaN, NaN, NaN,NaN,NaN)
+        return (NaN, NaN, NaN, NaN, NaN, NaN, NaN, NaN)
     end
 
-    a_f = f_ν*a/(2-ξ)
+    a_f = f_ν*a_i/(2-ξ)
+    #println(a_f)
 
     Lvec_norm = sqrt(α^2*sinθ^2*sinϕ^2+(h_ν*α*sinθ*cosϕ-j_ν*(1+α*cosθ))^2)
-    η = f_ν*g_ν^2*M/Mf*Lvec_norm^2
+    η = f_ν*g_ν^2*M_i/M_f*Lvec_norm^2
 
     e_f = sqrt(1+(ξ-2)*η)
 
     #angle between x and direction of motion
     # perhaps can be skipped to save some time, just get directly cos and sin
-    τ = acos(max(-1,min(1,j_ν)))
-    if (sinν < 0)
+    τ = acos(max(-1, min(1, j_ν)))
+    if (sinν_i < 0)
         τ = -τ
     end
     #δ = ω + ν - τ
     cosτ = cos(τ)
     sinτ = sin(τ)
-    cosδ = cosν*sinτ*sinω-sinν*cosτ*sinω+sinν*sinτ*cosω+cosν*cosτ*cosω
-    sinδ = sinν*sinτ*sinω+cosν*cosτ*sinω-cosν*sinτ*cosω+sinν*cosτ*cosω
+    cosδ = cosν_i*sinτ*sinω_i - sinν_i*cosτ*sinω_i + sinν_i*sinτ*cosω_i + cosν_i*cosτ*cosω_i
+    sinδ = sinν_i*sinτ*sinω_i + cosν_i*cosτ*sinω_i - cosν_i*sinτ*cosω_i + sinν_i*cosτ*cosω_i
 
     # Elements of rotation matrix
-    R_e_par =cosi*cosΩ*cosδ-sinΩ*sinδ
-    R_e_per = -(cosi*cosΩ*sinδ)-sinΩ*cosδ
-    R_e_z = -(sini*cosΩ)
-    R_n_par = cosΩ*sinδ+cosi*sinΩ*cosδ
-    R_n_per = cosΩ*cosδ-cosi*sinΩ*sinδ
-    R_n_z = -(sini*sinΩ)
-    R_o_par = sini*cosδ
-    R_o_per = -(sini*sinδ)
-    R_o_z = cosi
+    # RTW: this coordinate system uses E, N, and O, but the text refers instead to celestial W, not E... need to double check
```

Comment:
The text is right, this needs to be corrected. I had it wrong in the text before too. If you look to the sky, with celestial north pointing upwards, then celestial east points to the left.